src/main/java/seedu/clinic/logic/commands/EditCommand.java

```diff
@@ -100,89 +120,93 @@ public EditCommand(Index index, EditDescriptor editDescriptor) {
         this.index = index;
         if (editDescriptor instanceof EditSupplierDescriptor) {
             this.editDescriptor = new EditSupplierDescriptor((EditSupplierDescriptor) editDescriptor);
-            logger.log(Level.INFO, "Received instructions to edit supplier");
+            logger.log(Level.INFO, LOG_MESSAGE_RECEIVE_EDIT_SUPPLIER);
         } else {
-            assert editDescriptor instanceof EditWarehouseDescriptor : "editDescriptor supplied should be "
-                    + "of EditWarehouseDescriptor type here.";
+            assert editDescriptor instanceof EditWarehouseDescriptor : INVALID_EDIT_SUPPLIER_DESCRIPTOR_ASSERTION;
             this.editDescriptor = new EditWarehouseDescriptor((EditWarehouseDescriptor) editDescriptor);
-            logger.log(Level.INFO, "Received instructions to edit warehouse");
+            logger.log(Level.INFO, LOG_MESSAGE_RECEIVE_EDIT_WAREHOUSE);
         }
 
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Supplier> lastShownSupplierList = model.getFilteredSupplierList();
-        List<Warehouse> lastShownWarehouseList = model.getFilteredWarehouseList();
-        CommandResult commandResult;
 
         if (this.editDescriptor instanceof EditSupplierDescriptor) {
-            if (index.getZeroBased() >= lastShownSupplierList.size()) {
-                throw new CommandException(Messages.MESSAGE_INVALID_SUPPLIER_DISPLAYED_INDEX);
-            }
+            return executeSupplierEditing(model);
+        } else {
+            return executeWarehouseEditing(model);
+        }
```

Comment:
Else clause can be excluded here!

Reply:
I guess it's fine. I didn't exclude the Else because it looks more clear visually .